### PR TITLE
DT-995 reactor netty version upgrade

### PR DESCRIPTION
**Pull request by mikehalmamoj**
Overrides the bad version of reactor-netty imported by the Spring Boot Plugin.  And some version upgrades etc.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/DependencyManagementPluginManagerTest.kt [hunk 1]
```diff
@@ -22,6 +43,83 @@ class DependencyManagementPluginManagerTest : GradleBuildTest() {
...
+
+private fun kotlinWrongExplicitReactorNettyVersion(): String {
+  return """
+    plugins {
+      id("uk.gov.justice.hmpps.gradle-spring-boot") version "0.1"
+    }
+    dependencies {
+      implementation("io.projectreactor.netty:reactor-netty:0.9.8.RELEASE")
+    }
+  """.trimIndent()
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/DependencyManagementPluginManagerTest.kt:124]: Do we need  test where webflux is not a dependency (nor is netty) and assert that the jar is not added as a dependency?

Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/gradle/functional/pluginmanagers/DependencyManagementPluginManagerTest.kt:124]: Yes I think we do.  I've got a failing test to deal with anyway so I'll stick it in.

File: release-notes/0.4.3.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# 0.4.2
```
Comment (petergphillips) [inline release-notes/0.4.3.md:1]: 0.4.3

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5.1-bin.zip
```
Comment (petergphillips) [inline gradle/wrapper/gradle-wrapper.properties:3]: probably this that causes the tests to now fail

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt [hunk 4]
```diff
@@ -18,4 +19,13 @@ class DependencyManagementPluginManager(override val project: Project) : PluginM
       it.mavenBom(SpringBootPlugin.BOM_COORDINATES)
     }
   }
+
+  private fun forceTransitiveVersions(project: Project) {
+    val depManConfigurer = project.extensions.getByName("dependencyManagement") as DependencyManagementConfigurer
+    depManConfigurer.dependencies {
+      // Overriding 0.9.8.RELEASE introduced by Spring Boot 2.3.1 - remove when Spring Boot upgrades to > 0.9.8.RELEASE (along with tests in functional/DependencyManagementPluginManagerTest)
+      it.dependency("io.projectreactor.netty:reactor-netty:0.9.9.RELEASE")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt:27]: not understanding this.  Aren't you just always adding netty here?  What about projects that don't have it as a dependency?

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (mikehalmamoj) [reply gradle/wrapper/gradle-wrapper.properties]: Still stuck at 6.3 then.  Might have to raise a ticket with Gradle.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt (hunk 4 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt:27]: No, this works in the same way as resolution strategy forcing - added a test to prove this is true.

The code above doesn't add the dependency to any specific configuration, it just registers the version with the dependency management configurer.  So I guess that is doing the clever stuff under the bonnet.

Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt:27]: :+1:

**Pull request merged**